docs/using.md

```diff
@@ -0,0 +1,282 @@
+
+The following sections will explore these methods of data processing in more detail.
+
+## Topics, streams, partitions, replicas, and consumer groups
+
+The main structure used for data organization in Quix is the topic. For example, the topic might be `iot-telemetry`. To allow for horizontal scaling, a topic is typically divided into multiple streams. You may have multiple devices, or sources, writing data into a topic, so to ensure scaling and message ordering, each source writes into its own stream. Device 1 would write to stream 1, and device 2 to stream 2 and so on. This is the idea of [stream context](./features/streaming-context.md). 
+
+Quix Streams ensures that stream context is preserved, that is, messages inside one stream are always published to the same single partition. This means that inside one stream, a consumer can rely on the order of messages. A partition can contain multiple streams, but a stream is always confined to one partition.
+
+It is possible to organize the code that processes the streams in a topic using the idea of a consumer group. This indicates to the broker that you will process the topic with all available replicas.
+
+Horizontal scaling occurs automatically, because when you deploy multiple replicas, a stream is assigned to a replica. For example, if there are three streams and three replicas, each replica will process a single stream. If you had only one replica, it would need to process all streams in that topic. If you have three streams and two replicas, one replica would process two streams, and the other replica a single stream.
+
+When you create the consumer you specify the consumer group as follows:
+
+```python
+topic_consumer = client.get_topic_consumer(os.environ["input"], consumer_group = "empty-transformation")
+```
+
+!!! note
+
+    If you don't specify a consumer group, then all messages in all streams in a topic will be processed by all replicas in the microservice deployment.
+
+## Stream data formats
+
+There are two main formats of stream data: 
+
+1. **Event data** - in Quix Streams this is represented with the `qx.EventData` class.
+2. **Time-series** - in Quix Streams this is represented with the `qx.TimeseriesData` class (and two other classes: one for Pandas data frame format, and one for raw Kafka data).
+
+Event data refers to data that is independent, whereas time-series data is a variable that changes over time. An example of event data is a financial transaction. It contains all data for the invoice, with a timestamp (the time of the transaction), but a financial transaction itself is not a variable you'd track over time. The invoice may itself contain time-series data though, such as the customer's account balance. 
+
+Time-series data is a variable that is tracked over time, such as temperature from a sensor, or the g-forces in a racing car.
+
+Time-series data has three formats in Quix Streams:
```

Comment:
Time-series data has three different representations in Quix Streams to serve different use cases and developers. The underling data that these 3 models represent are the same.

Reply:
Fixed.